pkg/reconciler/config/store.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"reflect"
+	"sync/atomic"
+
+	"github.com/knative/pkg/configmap"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type Logger interface {
+	Infof(string, ...interface{})
+	Fatalf(string, ...interface{})
+	Errorf(string, ...interface{})
+}
+
+type UntypedStore struct {
+	name   string
+	logger Logger
+
+	storages     map[string]*atomic.Value
+	constructors map[string]reflect.Value
+}
+
+func NewUntypedStore(
+	name string,
+	logger Logger,
+	constructors ...interface{}) *UntypedStore {
+
+	store := &UntypedStore{
+		name:         name,
+		logger:       logger,
+		storages:     make(map[string]*atomic.Value),
+		constructors: make(map[string]reflect.Value),
+	}
+
+	// TODO(dprotaso) Check argument validity
+	for i := 0; i < len(constructors); i = i + 2 {
```

Comment:
The check in it's current form allows the code to panic when args are missing.

The suggested check doesn't do that
ie this would be acceptable input
```
NewUntypedStore(name, logger, "some-name")
```

Was there another reason for the different bounds check?

Reply:
Switching to a map makes this a non-issue